internal/health/condition/check_ready.go

```diff
@@ -48,6 +85,61 @@ func (r *readyCheck) Check(_ context.Context, etcd druidv1alpha1.Etcd) Result {
 		}
 	}
 
+	// Look for split-quorum/split-brain scenario.
+	leaseNames := druidv1alpha1.GetMemberLeaseNames(etcd.ObjectMeta, etcd.Spec.Replicas)
+	leases := make([]*coordinationv1.Lease, 0, len(leaseNames))
+	for _, leaseName := range leaseNames {
+		lease := &coordinationv1.Lease{}
+		if err := r.client.Get(ctx, client.ObjectKey{Namespace: etcd.Namespace, Name: leaseName}, lease); err != nil {
```

Comment:
if `err != nil` does not mean that it is not found. You have to explicitly check for `NotFoundError` If it is any other error then you should indicate that at this time you are unable to determine the condition.